[#160634397]f t-create-endpoint-for-user-signup

Johnpaul-Borree opened the pull request:
#### What does this PR do?
Create an endpoint for signup
#### Description of Task to be completed?
- Rearrange Folders
- Refactor codes to use es6 classes
- Refactor codes to post to a database
- create the model to query the database
- Create controller actions to enable the user to signup
- signup user with jwt.
- create a route to access the database through the controller
#### How should this be manually tested?
- npm run start creates a table by running migration script in package.json
- Test endpoint on postman
#### What are the relevant pivotal tracker stories?
[Signup to a database](https://www.pivotaltracker.com/story/show/160634397)
#### Screenshots (if appropriate)
![screenshot 21](https://user-images.githubusercontent.com/25976032/45862218-a8bce400-bd68-11e8-9b5d-f51657cac559.png)

Johnpaul-Borree committed:
[#160634397]create endpoint for user signup

- Refactor codes to use es6 classes
- Refactor codes to post to a database
- create model to query database
- Create controller actions to enable user to signup
- create route to access the database through the controller

KaiserPhemi commented on src/helpers/validations/userValidator.js:
```diff
@@ -0,0 +1,60 @@
+import { check, validationResult } from 'express-validator/check';
```
#### Critical
I believe you were told not to use `express-validator`. Why use it here??

Johnpaul-Borree replied:
Serious, I will take care of that and repeat validations, thanks

KaiserPhemi reviewed (changes requested):
Implement feedback & fix your failing tests.

Johnpaul-Borree committed:
[ft-#160634397] Implement feedback on signup

- Remove express validator module
- Add custom validation for sign up
- Use Jsdocs for comments
